csv/src/fs2/data/csv/package.scala

```diff
@@ -28,17 +29,45 @@ package object csv {
 
   type DecoderResult[T] = Either[DecoderError, T]
 
+  type HeaderResult[T] = Either[HeaderError, T]
+
+  type CsvNelRow[HeadElem] = CsvRow[NonEmptyList[HeadElem]]
+
+  type CsvNelRowDecoder[T, HeadElem] = CsvRowDecoder[T, NonEmptyList[HeadElem]]
+
+  object CsvNelRowDecoder {
+    def apply[T, HeadElem](implicit dec: CsvNelRowDecoder[T, HeadElem]): CsvNelRowDecoder[T, HeadElem] = dec
+  }
+
+  type ParseableNelHeader[HeadElem] = ParseableHeader[NonEmptyList[HeadElem]]
+
+  object ParseableNelHeader {
+    def apply[HeadElem](implicit dec: ParseableNelHeader[HeadElem]): ParseableNelHeader[HeadElem] = dec
+  }
+
   /** Transforms a stream of characters into a stream of CSV rows.
     */
   def rows[F[_]](separator: Char = ',')(
       implicit F: ApplicativeError[F, Throwable]): Pipe[F, Char, NonEmptyList[String]] =
     RowParser.pipe[F](separator)
 
+  /** Transforms a stream of strings into a stream of CSV rows. Helper for streams which are not initially char-based,
+   *  prefer to use [[rows]] otherwise.
+   */
+  def rowsFromStrings[F[_]](separator: Char = ',')(
+    implicit F: ApplicativeError[F, Throwable]): Pipe[F, String, NonEmptyList[String]] =
+    _.flatMap(s => Stream.chunk(Chunk.charBuffer(JCharBuffer.wrap(s)))) through RowParser.pipe[F](separator)
```

Comment:
Can you use the dot syntax please? Not a big fan of the dotless application :)

Reply:
Also is it like really way more efficient to use NIO than just `Stream.emits(s)`? If the performance gain is marginal, then I would prefer not having a `CharBuffer`.